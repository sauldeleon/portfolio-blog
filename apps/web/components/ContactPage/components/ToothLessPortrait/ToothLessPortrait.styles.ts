import Image from 'next/image'
import { css, keyframes, styled } from 'styled-components'

import { randomIntFromInterval } from '@web/utils/random'

import { portraitStyles } from '../Portrait/Portrait.styles'

const TEETH_DELAY_FACTOR = 5

/**                           
 *   
  Each 2 values, it doubles the previous increment
  t(0) = 5*1 + 0 = 5
  t(1) = 5*1 + t(n-1) = 10 
  t(2) = 5*2 + t(n-1) = 20
  t(3) = 5*2 + t(n-1) = 30
  t(4) = 5*4 + t(n-1) = 50
  t(5) = 5*4 + t(n-1) = 70
  t(6) = 5*8 + t(n-1) = 110
  t(7) = 5*8 + t(n-1) = 150

  t(0) = 5
  t(n) = t(n - 1) + 5 * 2^(n // 2)
*/
export const getDelay = (n: number): number => {
  if (n === 0) {
    return TEETH_DELAY_FACTOR
  } else {
    return getDelay(n - 1) + TEETH_DELAY_FACTOR * Math.pow(2, Math.floor(n / 2))
  }
}

export const getWeightedDelay = (delay: number, totalDelay: number) =>
  (delay * 100) / totalDelay

export const getAdditiveBackgroundUrl = (images: string[], index: number) =>
  images
    .slice(0, index + 1)
    .map((image) => `url(${image})`)
    .join(', ')

export const generateKeyframeStep = ({
  images,
  totalImagesDelay,
  index,
}: {
  images: string[]
  totalImagesDelay: number
  index: number
}) => {
  const weightedDelay = getWeightedDelay(getDelay(index), totalImagesDelay)
  return css`
    ${weightedDelay - 0.0001}% {
      background-image: ${index > 0
        ? getAdditiveBackgroundUrl(images, index - 1)
        : 'none'};
    }

    ${weightedDelay}% {
      background-image: ${getAdditiveBackgroundUrl(images, index)};
    }
  `
}

const swapImageAnimation = (
  images: string[],
  totalImagesDelay: number
) => keyframes`
      0% {
        background-image: none;
      }
      ${images.map((_, index) =>
        generateKeyframeStep({ images, totalImagesDelay, index })
      )}
`

export const StyledPortraitContainer = styled.div`
  position: relative;
  z-index: -1;
  height: 70px;
  width: 70px;
  overflow: hidden;

  ${({ theme }) => theme.media.up.md} {
    height: 140px;
    width: 140px;
  }
`

export const StyledToothLessPortrait = styled(Image)`
  ${portraitStyles};
`

export const RotateTooth = styled.div`
  position: absolute;
  animation: 2s ${({ theme }) => theme.animation.moveToothRotate} 1s linear
    infinite;
  top: 50px;
  left: 49.1%;
  transform: rotate(180deg);
`

interface StyledToothProps {
  $p1h?: number
  $p1v?: number
  $p2h?: number
  $p2v?: number
  $cp1h?: number
  $cp1v?: number
  $cp2h?: number
  $cp2v?: number
  $index: number
}

export const StyledTooth = styled.div<StyledToothProps>`
  ${({
    $p1h = 0,
    $p1v = 0,
    $p2h = -558,
    $p2v = 220,
    $cp1h = 0,
    $cp1v = 250,
    $cp2h = -280,
    $cp2v = 200,
    $index,
    theme,
  }) => css`
    position: absolute;
    width: 100%;
    height: 100%;
    top: -200%;
    left: -6px;
    animation-name: ${theme.animation.moveToothFall};
    animation-duration: 5s;
    animation-delay: ${getDelay($index)}s;
    animation-iteration-count: 1;

    ${theme.media.up.md} {
      --p1h: ${$p1h}; //initial x position
      --p1v: ${$p1v}; //initial y position
      --p2h: ${$p2h}; //final x position
      --p2v: ${$p2v}; //final y position
      --cp1h: ${$cp1h}; //initial x control curve position
      --cp1v: ${$cp1v}; //initial y control curve position
      --cp2h: ${$cp2h}; //final x control curve position
      --cp2v: ${$cp2v}; //final y control curve position
      --final-tooth-top: ${randomIntFromInterval(
        10,
        90,
        `tooth-height-${$index}`
      )}%;

      top: 0;
      left: -70%;
      animation-name: ${theme.animation.moveToothHorizontal},
        ${theme.animation.moveToothVertical};
      animation-timing-function: cubic-bezier(
          calc(1 / 3),
          calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),
          calc(2 / 3),
          calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))
        ),
        cubic-bezier(
          calc(1 / 3),
          calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),
          calc(2 / 3),
          calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))
        );
    }
  `}
`

export const ToothHoleImage = styled.div<{
  $images: string[]
}>`
  position: absolute;
  background-image: none;
  background-size: contain;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  ${({ $images }) => {
    const totalImagesDelay = getDelay($images.length - 1)

    return css`
      animation: ${totalImagesDelay}s
        ${swapImageAnimation($images, totalImagesDelay)} linear forwards;
    `
  }}
`
