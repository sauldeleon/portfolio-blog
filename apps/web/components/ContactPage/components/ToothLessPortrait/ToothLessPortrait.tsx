import { useId } from 'react'

import { Tooth } from '@sdlgr/assets'

import { useClientTranslation } from '@web/i18n/client'

import {
  RotateTooth,
  StyledPortraitContainer,
  StyledTooth,
  StyledToothLessPortrait,
  ToothHoleImage,
} from './ToothLessPortrait.styles'

interface ToothLessPortraitProps {
  onClick: () => void
}

export function ToothLessPortrait({ onClick }: ToothLessPortraitProps) {
  const { t } = useClientTranslation('contactPage')

  const id = useId()
  const toothImages = [
    '/assets/toothPlace-1.png',
    '/assets/toothPlace-2.png',
    '/assets/toothPlace-3.png',
    '/assets/toothPlace-4.png',
    '/assets/toothPlace-5.png',
    '/assets/toothPlace-6.png',
    '/assets/toothPlace-7.png',
    '/assets/toothPlace-8.png',
  ]

  return (
    <div onClick={onClick}>
      {toothImages.map((_, index) => (
        <StyledTooth key={`${id}-${index}`} $index={index}>
          <RotateTooth>
            <Tooth width={10} height={10} />
          </RotateTooth>
        </StyledTooth>
      ))}

      <StyledPortraitContainer>
        <StyledToothLessPortrait
          src="/assets/portrait.jpg"
          alt={t('toothlessProfilePicture')}
          width={140}
          height={140}
        />
        <ToothHoleImage $images={toothImages} />
      </StyledPortraitContainer>
    </div>
  )
}
