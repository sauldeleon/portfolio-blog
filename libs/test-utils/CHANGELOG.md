# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
