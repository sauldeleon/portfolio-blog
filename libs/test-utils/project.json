{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
