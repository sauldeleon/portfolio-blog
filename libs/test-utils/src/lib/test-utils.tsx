import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
