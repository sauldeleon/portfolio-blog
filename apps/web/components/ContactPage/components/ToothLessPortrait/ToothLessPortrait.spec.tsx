import { screen } from '@testing-library/react'

import { renderApp } from '@sdlgr/test-utils'

import { ToothLessPortrait } from './ToothLessPortrait'

describe('ToothLessPortrait', () => {
  it('should render', async () => {
    const { baseElement } = renderApp(
      <ToothLessPortrait onClick={() => void 0} />
    )
    await screen.findByAltText('My toothless profile picture')
    expect(baseElement).toMatchSnapshot()
  })
})
