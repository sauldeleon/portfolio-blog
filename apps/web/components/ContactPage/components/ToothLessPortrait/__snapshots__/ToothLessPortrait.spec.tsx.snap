// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToothLessPortrait should render 1`] = `
.c9 {
  position: relative;
  z-index: -1;
  height: 70px;
  width: 70px;
  overflow: hidden;
}

.c10 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
}

.c1 {
  position: absolute;
  animation: 2s bOWqww 1s linear infinite;
  top: 50px;
  left: 49.1%;
  transform: rotate(180deg);
}

.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 5s;
  animation-iteration-count: 1;
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 10s;
  animation-iteration-count: 1;
}

.c3 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 20s;
  animation-iteration-count: 1;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 30s;
  animation-iteration-count: 1;
}

.c5 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 50s;
  animation-iteration-count: 1;
}

.c6 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 70s;
  animation-iteration-count: 1;
}

.c7 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 110s;
  animation-iteration-count: 1;
}

.c8 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: -200%;
  left: -6px;
  animation-name: kAWDoo;
  animation-duration: 5s;
  animation-delay: 150s;
  animation-iteration-count: 1;
}

.c11 {
  position: absolute;
  background-image: none;
  background-size: contain;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  animation: 150s dXIgQ linear forwards;
}

@media (width >= 768px) {
  .c9 {
    height: 140px;
    width: 140px;
  }
}

@media (width >= 768px) {
  .c10 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c0 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 13%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c2 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 43%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c3 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 12%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c4 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 26%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c5 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 71%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c6 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 63%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c7 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 45%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

@media (width >= 768px) {
  .c8 {
    --p1h: 0;
    --p1v: 0;
    --p2h: -558;
    --p2v: 220;
    --cp1h: 0;
    --cp1v: 250;
    --cp2h: -280;
    --cp2v: 200;
    --final-tooth-top: 31%;
    top: 0;
    left: -70%;
    animation-name: ifoKcn,bfulbY;
    animation-timing-function: cubic-bezier(           calc(1 / 3),           calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),           calc(2 / 3),           calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))         ),cubic-bezier(           calc(1 / 3),           calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),           calc(2 / 3),           calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))         );
  }
}

<body>
  <div>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c1"
        >
          <svg
            height="10"
            width="10"
          >
            tooth.svg
          </svg>
        </div>
      </div>
      <div
        class="c9"
      >
        <img
          alt="My toothless profile picture"
          class="c10"
          data-nimg="1"
          decoding="async"
          height="140"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fportrait.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fassets%2Fportrait.jpg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="140"
        />
        <div
          class="c11"
        />
      </div>
    </div>
  </div>
</body>
`;
